keps/sig-scheduling/2926-job-mutable-scheduling-directives/README.md

```diff
@@ -0,0 +1,874 @@
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Most kubernetes components are pod centric, including the scheduler and cluster 
+autoscaler. This works well for service type workloads where the pods of a service
+are mostly independent and all services are expected to be running at all times.
+
+However, for batch jobs, it doesn't make sense to focus only on pods. In most cases
+a parallel job will want the pods to run within specific constraints, like all in 
+the same zone, or all either on GPU model x or y, but not a mix of both.
+
+We made the first step towards achieving those semantics by introducing the `suspend`
+flag to the Job API, which allowed a queue controller to decide when a job should start.
+The idea was that such a controller would unsuspend the job when specific constraints
+are met (e.g., capacity is provisioned). However, once a job is unsuspended, the queue 
+controller has no influence on where the pods of a job will actually land.
+
+Adding the ability to mutate a job's scheduling bits before it starts gives a queue controller
+the ability to influence placement while at the same time offloading actual pod-to-node
+assignment to the existing kube-scheduler.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- Allow mutating node affinity, node selector and tolerations of jobs that 
+  have never been unsuspended.
+
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- Implement a queue controller.
+- Allow mutating scheduling bits of jobs that started at least once.
+- Allow mutating scheduling bits of pods.
+- Allow mutating scheduling bits of apps other than jobs.
+
```

Comment:
You mentioned somewhere to support preemptible job, I guess now it's not supported. If so, maybe document here as a non-goal?

Reply:
added